docs/data-sources/config.md

```diff
@@ -0,0 +1,66 @@
+---
+page_title: "cloudinit_config Data Source - terraform-provider-cloudinit"
+description: |-
+  Renders a multi-part MIME configuration https://cloudinit.readthedocs.io/en/latest/explanation/format.html#mime-multi-part-archive for use with cloud-init https://cloudinit.readthedocs.io/en/latest/.
+  Cloud-init is a commonly-used startup configuration utility for cloud compute instances. It accepts configuration via provider-specific user data mechanisms, such as user_data for Amazon EC2 instances. Multi-part MIME is one of the data formats it accepts. For more information, see User-Data Formats https://cloudinit.readthedocs.io/en/latest/explanation/format.html in the cloud-init manual.
+  This is not a generalized utility for producing multi-part MIME messages. It's feature set is specialized for cloud-init multi-part MIME messages.
+---
+
+# cloudinit_config (Data Source)
+
+Renders a [multi-part MIME configuration](https://cloudinit.readthedocs.io/en/latest/explanation/format.html#mime-multi-part-archive) for use with [cloud-init](https://cloudinit.readthedocs.io/en/latest/).
+
+Cloud-init is a commonly-used startup configuration utility for cloud compute instances. It accepts configuration via provider-specific user data mechanisms, such as `user_data` for Amazon EC2 instances. Multi-part MIME is one of the data formats it accepts. For more information, see [User-Data Formats](https://cloudinit.readthedocs.io/en/latest/explanation/format.html) in the cloud-init manual.
+
+This is not a generalized utility for producing multi-part MIME messages. It's feature set is specialized for cloud-init multi-part MIME messages.
+
+## Example Usage
+
+```terraform
+data "cloudinit_config" "foobar" {
+  gzip          = false
+  base64_encode = false
+
+  part {
+    content = "foo"
+
+    content_type = "text/x-shellscript"
+    filename     = "foo.sh"
+  }
+
+  part {
+    content = "bar"
+
+    content_type = "text/x-shellscript"
+    filename     = "bar.sh"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `base64_encode` (Boolean) Specify whether or not to base64 encode the `rendered` output. Defaults to `true`, and cannot be disabled if gzip is `true`.
+- `boundary` (String) Specify the Writer's default boundary separator. Defaults to `MIMEBOUNDARY`.
+- `gzip` (Boolean) Specify whether or not to gzip the `rendered` output. Defaults to `true`.
+- `part` (Block List) A nested block type which adds a file to the generated cloud-init configuration. Use multiple `part` blocks to specify multiple files, which will be included in order of declaration in the final MIME document. (see [below for nested schema](#nestedblock--part))
```

Comment:
**Note:** I think blocks are determined as `Required` by the `MinItems` field [here](https://github.com/hashicorp/terraform-plugin-docs/blob/b169348cd191d12f7ef3a94da5b4716d0aed785f/schemamd/behaviors.go#L11), which the plugin framework explicitly [does not have](https://github.com/hashicorp/terraform-plugin-framework/blob/faab4e35bd4e13c6f0bff2c8223c9c52996304c2/internal/fwschema/block.go#L21)

I'm not sure what my options are for modifying the docs/schema to make this not confusing for anyone trying to use the provider.

Reply:
I didn't see an existing feature request issue for this over in terraform-plugin-docs, so might be good to create one to capture the use case.

In the meantime, an option to potentially make this less confusing is to switch this to a manual template, rather than using the default template and its specific schema output behaviors.